Shared/Samples/Validate utility network topology/ValidateUtilityNetworkTopologyView.Model.swift

```diff
@@ -0,0 +1,448 @@
+            Tap 'Get State' to check the updated network state.
+            """
+        }
+        
+        /// Selects a feature from a given list of identify layer results.
+        /// - Parameter identifyResults: The identify layer results.
+        func selectFeature(from identifyResults: [IdentifyLayerResult]) {
+            clearSelection()
+            
+            // Get the first feature from the results.
+            let layerResult = identifyResults.first {
+                let layerName = $0.layerContent.name
+                return layerName == .deviceTableName || layerName == .lineTableName
+            }
+            guard let feature = layerResult?.geoElements.first as? ArcGISFeature else { return }
+            
+            // Get the coded values from the feature's field.
+            let fieldName: String = feature.table?.tableName == .deviceTableName
+            ? .deviceStatusField
+            : .nominalVoltageField
+            
+            guard let field = feature.table?.field(named: fieldName) else { return }
+            self.field = field
+            
+            guard let codedValues = (field.domain as? CodedValueDomain)?.codedValues else { return }
+            fieldValueOptions = codedValues
+            
+            // Get the current attribute value from the feature.
+            let fieldValue = feature.attributes[field.name]
+            selectedFieldValue = codedValues.first { valuesAreEqual($0.code, fieldValue) }
+            
+            // Select the identified feature.
+            let featureLayer = feature.table?.layer as? FeatureLayer
+            featureLayer?.selectFeature(feature)
+            canClearSelection = true
+            self.feature = feature
+            
+            statusMessage = "Select a new '\(field.alias)'."
+        }
+        
+        /// Applies the edits to the feature to the service.
+        func applyEdits() async throws {
+            guard let feature,
+                  let serviceFeatureTable = feature.table as? ServiceFeatureTable,
+                  let serviceGeodatabase = serviceFeatureTable.serviceGeodatabase,
+                  let fieldName = field?.name.trimmingCharacters(in: .whitespacesAndNewlines)
+            else { return }
+            
+            // Update the feature with the new value in the it's feature table.
+            statusMessage = "Updating feature..."
+            feature.setAttributeValue(selectedFieldValue?.code, forKey: fieldName)
+            try await serviceFeatureTable.update(feature)
+            
+            // Apply the edits in the feature table to the service.
+            statusMessage = "Applying edits..."
+            let featureTableEditResults = try await serviceGeodatabase.applyEdits()
+            
+            // Determine if the attempt to edit resulted in any errors.
+            let didNotCompleteWithErrors = featureTableEditResults.allSatisfy { tableEditResult in
```

Comment:
I think this name can be read 2 ways and so maybe something like `didCompleteSuccessfully` is clearer?